src/common/components/elements/ListingPreview.tsx

```diff
@@ -0,0 +1,343 @@
+import { Storefront } from '@/modules/storefront/types';
+import { Skeleton, Card, Row, Image, Typography } from 'antd';
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { DateTime, Duration } from 'luxon';
+import { DemoStorefront } from '@/common/constants/demoContent';
+import { ZoomInOutlined } from '@ant-design/icons';
+const { Title, Text } = Typography;
+
+interface Creator {
+  address: string;
+  verified?: boolean;
+  share: number;
+}
+
+interface Bid {
+  canceled: boolean;
+  last_amount: number;
+  last_changed_at: string; // ISO Date
+}
+
+interface Item {
+  address: string;
+  name: string;
+  uri: string;
+}
+
+interface NFTMetadata {
+  description: string;
+  external_url: string;
+  image: string; // url to image, often ipfs, sometimes arweave
+  name: string;
+  seller_fee_basis_points: number; // in thousands. Prbably need to divide by 100
+  symbol: string;
+  properties: {
+    category: 'image' | string;
+    creators: Creator[];
+    files: {
+      type: 'image/gif' | string;
+      uri: string; // arweave URI
+    }[];
+  };
+}
+
+export interface Listing {
+  address: string;
+  ends_at?: string | null; // ISO Date
+  created_at: string; // ISO Date
+  ended: boolean;
+  last_bid?: number | null; // lamport price of last bid (thought it was a timestamp at first)
+  price_floor?: number | null;
+  total_uncancelled_bids?: number | null;
+  instant_sale_price?: number | null; // is often 10000000000
+  subdomain: string;
+  storeTitle: string;
+  items: Item[]; // NFT metadata, including URI to fetch more data
+  nftMetadata?: NFTMetadata[]; // same length as items. Is set on mount
+}
```

Comment:
This issue is my fault, but I want to get all these fields changed to camelCase names on the backend before launch.

Reply:
Done!